internal/pkg/v2/README.md

```diff
@@ -0,0 +1,296 @@
+## v2 Package
+
+[TOC]
+
+### Overview
+
+This package encapsulates the APIv2 implementation.
```

Comment:
This was super-helpful in understanding this PR. Thanks for including it, and the link to the related presentation!